### PR TITLE
Added ensure_resource and ensure_packages to avoid collisions

### PR DESCRIPTION
**Pull request by dannygoulder**
Needs newer version of puppetlabs-stdlib which I have added to the metadata.json.

Puppetlabs-stdlib 4.x [only works with Puppet 3.x](https://github.com/puppetlabs/puppetlabs-stdlib/tree/4.1.0#compatibility) which is already a minimum requirement for this module.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -1,20 +1,18 @@
 # See README.md for usage information
 class sssd::install (
+  $ensure                = $sssd::ensure,
```
Comment (edestecd) [inline manifests/install.pp]: I don't see why this change is necessary.  The old variable name seems fine.
I would't want to break anyone's configuration, that is currently using this class directly...

File: manifests/init.pp [hunk 2]
```diff
@@ -123,7 +123,9 @@
   validate_re($service_ensure, '^running|true|stopped|false$')
 
   anchor { 'sssd::begin': } ->
-  class { '::sssd::install': } ->
+  class { '::sssd::install':
+    ensure => $sssd_package_ensure,
```
Comment (edestecd) [inline manifests/init.pp]: I think that reading this var from the install class is cleaner...
Can we leave this code alone?  Is there a reason to change it?

Comment (dannygoulder) [inline manifests/init.pp]: my local tests were failing because the var wasn't being read from the install class.  Is it a scoping issue with my version of Puppet (4.5)?  Agree that it's cleaner that way.  Maybe i can work around it in another way

File: manifests/install.pp [hunk 3]
```diff
@@ -6,7 +6,7 @@
   $extra_packages_ensure = $sssd::extra_packages_ensure,
 ) {
 
-  ensure_packages([$sssd_package],{ ensure => $ensure })
+  ensure_packages({$sssd_package => { 'ensure' => $ensure } } )
```
Comment (dannygoulder) [inline manifests/install.pp]: This is causing Puppet 3.8.7 with ruby 1.9.3 to fail in a strange way.  need to check on correct syntax as this works on ruby 2.1

File: manifests/install.pp (hunk 1 above)
Comment (dannygoulder) [inline manifests/install.pp]: Just realised my error here - think I merged more than I should have from the earlier PR. Also probably explains why the variable wasn't picked up above.

I'll clean that up tomorrow.

Comment (dannygoulder) [inline manifests/install.pp]: All cleaned up now.

**Comment by sgnl05** [comment]
This looks good too me. @edestecd do you have anything to add or should we just merge this?

**Comment by edestecd** [comment]
Everything looks great!
@dannygoulder  squashing all your commits to one or two commits would be nice, if you are comfortable with that...

Looks like github can squash for us, if you prefer...

**Comment by dannygoulder** [comment]
GitHub squash is a good plan. Cheers

**Comment by edestecd** [comment]
@dannygoulder thanks for the contribution!

**Comment by dannygoulder** [comment]
You're welcome. Thanks for the great work on the module.
